src/history/HistoryManagerImpl.cpp

```diff
@@ -317,6 +333,11 @@ HistoryManagerImpl::publishQueuedHistory()
     }
 #endif
 
+    if (!mApp.modeHasDatabase())
```

Comment:
meta question here...

I think we could simplify (and remove more of those `if (!mApp.modeHasDatabase())`) calls:
what if we were instead setting up an in memory sqlite database, that way code that deals with tables that are low churn like `PersistentState` would not have to be changed.
We could also not create the expensive tables at all, but we can also just check in tests that we don't populate them (and a bug in that front would be trivial to spot as we would have large memory usage).

Reply:
Possible! But will defer this question to a later PR as discussed in chat.